torch/fx/experimental/proxy_tensor.py

```diff
@@ -526,10 +526,25 @@ def wrapped(*args):
         fake_tensor_mode: Any = nullcontext()
         if tracing_mode == "real":
             fake_tensor_mode = nullcontext()
+
+            def restore_fake_tensor_mode():
+                return nullcontext()
         elif tracing_mode == "fake":
             fake_tensor_mode = FakeTensorMode(allow_fallback_kernels=True)
+            # initialize the mode
+            with fake_tensor_mode:
+                pass
+
+            def restore_fake_tensor_mode():
+                return fake_tensor_mode.restore()
```

Comment:
@samdow This feels really dumb. Maybe we should just let people reuse the mode object without calling restore? First time sets the ancestors, second time and after does a restore.

Reply:
Yeah that makes sense since [ProxyTensor had this too](https://github.com/pytorch/pytorch/pull/82549/files/bbeb147910362e51574ffcb2afa9cc68df6beabc#r935367338). Also we can probably just get rid of that since ProxyTensor is now mode only? I can put up a PR today unless you feel strongly

Reply:
Hmm, yes, I agree this can be simplified; it is probably best to wait till https://github.com/pytorch/pytorch/pull/83330 lands though, to avoid conflicts